sessions-core/src/main/java/com/adyen/checkout/sessions/core/SessionComponentCallback.kt

```diff
@@ -9,27 +9,175 @@
+     *
+     * You need to return [true] if you want to take over the sessions flow, otherwise the API calls will still be
+     * handled internally by the SDK. This could be useful in case you want to handle the flow yourself only in certain
+     * conditions, then you can return [false] if these conditions are not met.
+     *
+     * Once you take over the flow you will need to handle all the necessary subsequent network calls, otherwise a
+     * [MethodNotImplementedException] will be thrown.
+     *
+     * We provide a [PaymentMethodDetails] object that contains a non-serialized version of the partial payment method
+     * JSON. Use [PaymentMethodDetails.SERIALIZER] to serialize it to a [JSONObject].
+     *
+     * You should eventually call [component.resolveBalanceResult] with a [BalanceResult] containing the result of the
+     * network request.
+     *
+     * Note that not overriding this method while enabling partial payments will cause a [MethodNotImplementedException]
+     * to be thrown.
+     *
+     * See https://docs.adyen.com/api-explorer/ for more information on the API documentation.
+     *
+     * @param paymentMethodDetails The data from the partial payment method component.
+     * @return [true] if you took over the sessions flow, [false] otherwise.
+     */
     fun onBalanceCheck(paymentMethodDetails: PaymentMethodDetails): Boolean = false
+
+    /**
+     * Only applicable for partial payments flow.
+     *
+     * Override this method if you want to take over the sessions flow and make a network call to the /orders endpoint
+     * of the Checkout API through your server. This method is called when the user is trying to pay a part of the
+     * Drop-in amount using a partial payment method.
+     *
+     * You need to return [true] if you want to take over the sessions flow, otherwise the API calls will still be
+     * handled internally by the SDK. This could be useful in case you want to handle the flow yourself only in certain
+     * conditions, then you can return [false] if these conditions are not met.
+     *
+     * Once you take over the flow you will need to handle all the necessary subsequent network calls, otherwise a
+     * [MethodNotImplementedException] will be thrown.
+     *
+     * You should eventually call [component.resolveOrderResponse] with an [OrderResponse] containing the result of the
+     * network request.
+     *
+     * Note that not overriding this method while enabling partial payments will cause a [MethodNotImplementedException]
+     * to be thrown.
+     *
+     * See https://docs.adyen.com/api-explorer/ for more information on the API documentation.
+     *
+     * @return [true] if you took over the sessions flow, [false] otherwise.
+     */
     fun onOrderRequest(): Boolean = false
-    fun onOrderCancel(order: Order): Boolean = false
 
+    fun onOrderCancel(order: Order): Boolean = false // TODO remove this method?
```

Comment:
Why should we remove this method?

Reply:
It won't be applicable to standalone components

Reply:
Ahh right, we'll probably remove it then when working more on partial payments